src/main/java/games/strategy/engine/chat/Chat.java

```diff
@@ -55,6 +55,57 @@
     LOBBY_CHATROOM, GAME_CHATROOM, NO_SOUND
   }
 
+  public Chat(final IMessenger messenger, final String chatName, final IChannelMessenger channelMessenger,
+      final IRemoteMessenger remoteMessenger, final CHAT_SOUND_PROFILE chatSoundProfile) {
+
+    this.chatSoundProfile = chatSoundProfile;
+    this.messengers = new Messengers(messenger, remoteMessenger, channelMessenger);
+    statusManager = new StatusManager(messengers);
+    chatChannelName = ChatController.getChatChannelName(chatName);
+    this.chatName = chatName;
+    sentMessages = new SentMessagesHistory();
+
+    // the order of events is significant.
+    // 1 register our channel listener
+    // once the channel is registered, we are guaranteed that
+    // when we receive the response from our init(...) message, our channel
+    // subscriber has been added, and will see any messages broadcasted by the server
+    // 2 call the init message on the server remote. Any add or join messages sent from the server
+    // will queue until we receive the init return value (they queue since they see the init version is -1)
+    // 3 when we receive the init message response, acquire the lock, and initialize our state
+    // and run any queued messages. Queued messages may be ignored if the
+    // server version is incorrect.
+    // this all seems a lot more involved than it needs to be.
+    final IChatController controller = (IChatController) messengers.getRemoteMessenger()
+        .getRemote(ChatController.getChatControlerRemoteName(chatName));
+    messengers.getChannelMessenger().registerChannelSubscriber(m_chatChannelSubscribor,
+        new RemoteName(chatChannelName, IChatChannel.class));
+    final Tuple<Map<INode, Tag>, Long> init = controller.joinChat();
+    final Map<INode, Tag> chatters = init.getFirst();
+    synchronized (mutexNodes) {
```

Comment:
Nit: I know this was just copy-pasted from the old `init()` method, but synchronization shouldn't really be necessary in a constructor unless we expose the `this` reference outside the class before the constructor returns (a bad smell, in general).  I don't see that happening here, but I also didn't scrutinize it very deeply.  It may be too risky to muck with concurrency at this time, so I'm ok with leaving it as-is.